### PR TITLE
Delete project

### PR DESCRIPTION
**Pull request by grega**
Admin (soft) deletion of a project and its (non-owner) users, via the setting of a `deletedAt` timestamp (on both the project and its member / supervisor users).

If viewed, the project will be shown to have been deleted (exclusion of deleted projects from the admin listing will be handled in a new PR).

### CODE REVIEW

File: client/src/admin/projects/View.vue [hunk 1]
```diff
@@ -84,15 +91,43 @@
           <td>Member</td>
         </tr>
       </table>
+      <hr>
+      <h2>Admin Actions</h2>
+      <button class="btn btn-outline-danger" v-if="!project.deletedAt" v-on:click="confirmDeleteProject()">Delete Project</button>
+      <p v-else>No actions available</p>
```
Comment (Wardormeur) [inline client/src/admin/projects/View.vue]: Just chatted w/ @conoramurphy, deleted projects shouldn't show up in the admin project listing, so it shouldn't be possible to get to this page :p

Comment (conoramurphy) [reply client/src/admin/projects/View.vue]: Apologies for not specifying it on the ticket better.

Comment (grega) [reply client/src/admin/projects/View.vue]: Ah boo, I just made it all pretty ;)

screen shot 2018-04-11 at 10 25 38

**Comment by grega** [comment]
screen shot 2018-04-11 at 13 36 41

***

screen shot 2018-04-11 at 13 36 57

**Comment by grega** [comment]
@Wardormeur Yep totally understand, I would consider soft deletion via `update` useful _if_ `delete` was going to be used to hard delete, as there can be a use-case for both (and this was an assumption I made). I will happily pop it in the backlog to use `delete` (and override record destroying with soft-deletion / updating the record) instead.

**Comment by grega** [comment]
@Wardormeur done 👍

**Comment by grega** [comment]
Added a new task here for listing views w/ deleted projects: https://trello.com/c/7fQ5TWQW/289-exclude-deleted-projects-from-listings

**Pull request merged**